app/service/file_svc.py

```diff
@@ -34,30 +34,41 @@ def __init__(self, plugins, exfil_dir):
         except Exception as e:
             return web.HTTPNotFound(body=e)
 
-    async def upload(self, request):
+    async def upload(self, request, file_target=None, filebase=None, xored=False):
         """
         Accept a multipart file via HTTP and save it to the server
-        :param request:
-        :return: None
+        :param request: request object from aiohttp
+        :param file_target: filename of the file to be saved
+        :param filebase: base directory to save an uploaded file to
+        :param xored: whether the file needs to be encrypted on disk or not
+        :return: raw web response for failure or success
         """
         try:
             reader = await request.multipart()
-            exfil_dir = await self._create_exfil_sub_directory(request.headers)
+            save_dir = await self._create_exfil_sub_directory(request.headers)
```

Comment:
it's not clear to me what the changes in this function are intended to do, related to the listing of payloads? it looks like we're adding the functionality of changing name+base+xored but this seems disconnected from the list_payloads part of this PR.

Reply:
This is consolidating the functionality of `upload` to handle file upload from the GUI via `save_file` and files uploads from the agent via abilities.